### PR TITLE
FEATURE: Add HTTP Status Code to Shortcuts

### LINKED ISSUE #4410: FEATURE: Use something other than 303 for Neos Shortcut redirect
### Is there an existing issue for this topic?

- [X] I have searched the existing issues

### Description

Currently, the Neos Shortcut redirects with the HTTP Status code 303. 
Our SEO team suggest using 302 for those redirects for SEO reasons. 
Right now, the 303 status code is hard-coded into the redirect function. 

### Possible Solution

There are three possible solutions I can think of: 

- globally setting the HTTP Status code for Neos Shortcuts
- Hard coding a 302 HTTP Status code for Neos Shortcuts 
- Using a dropdown in the Neos UI to set the status code for each Neos Shortcut

### PR DESCRIPTION
**Pull request by pKallert**
resolves #4410 

According to the Poll here: https://github.com/neos/neos-development-collection/issues/4410#issuecomment-1867545755 we will be using a global Setting for the Shortcut instead of a nodeType.

**Checklist**

- [ ] Code follows the PSR-2 coding style
- [ ] Tests have been created, run and adjusted as needed
- [ ] The PR is created against the [lowest maintained branch](https://www.neos.io/features/release-roadmap.html)
- [ ] Reviewer - PR Title is brief but complete and starts with `FEATURE|TASK|BUGFIX`
- [ ] Reviewer - The first section explains the change briefly for change-logs
- [ ] Reviewer - Breaking Changes are marked with `!!!` and have upgrade-instructions

### CODE REVIEW

File: Neos.Neos/NodeTypes/Document/Shortcut.yaml [hunk 1]
```diff
@@ -46,3 +46,15 @@
...
+    httpStatusCode:
+      type: int
+      defaultValue: 303
+      ui:
+        label: i18n
+        inspector:
+          # change this to edit the status code in the inspector
+          hidden: true
+          group: 'document'
+          editor: 'Neos.Neos/Inspector/Editors/TextFieldEditor'
```
Comment (jonnitto) [inline Neos.Neos/NodeTypes/Document/Shortcut.yaml]: I would offer here a selectbox with the different values

File: Neos.Neos/Migrations/ContentRepository/Version20230809164806.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+comments: 'Migrate shortcut nodes: add httpStatusCode property'
+migration:
+  -
+    filters:
+      -
+        type: 'NodeType'
+        settings:
+          nodeType: 'Neos.Neos:Shortcut'
```
Comment (jonnitto) [inline Neos.Neos/Migrations/ContentRepository/Version20230809164806.yaml]: If someone create a own node type with the super type `Neos.Neos:Shortcut`, we still need to add this value

```suggestion
          nodeType: 'Neos.Neos:Shortcut'
          withSubTypes: true
```

**Review by jonnitto — CHANGES_REQUESTED** [review]

File: Neos.Neos/NodeTypes/Document/Shortcut.yaml (hunk 1 above)
Comment (pKallert) [reply Neos.Neos/NodeTypes/Document/Shortcut.yaml]: According to the Poll in the issue we will be using a general setting so I removed the whole thing

File: Neos.Neos/Migrations/ContentRepository/Version20230809164806.yaml (hunk 2 above)
Comment (pKallert) [reply Neos.Neos/Migrations/ContentRepository/Version20230809164806.yaml]: We will be using a general setting so the Migration is no longer needed

**Review by jonnitto — APPROVED** [review]
Fine. (by reading)

**Review by dlubitz — APPROVED** [review]
❤️ Thanks a lot for taking care

**Review by kitsunet — APPROVED** [review]

**Pull request merged**